filters/PMFFilter.cpp

```diff
@@ -105,7 +105,11 @@ std::vector<double> PMFFilter::morphOpen(PointViewPtr view, float radius)
     // dilate
     for (PointId i = 0; i < np; ++i)
```

Comment:
In the second pass through the points, you are comparing the current elevation to the eroded elevation of the neighbors, which is unknown until you are done with the first pass.